corehq/pillows/synclog.py

```diff
@@ -104,18 +104,22 @@ def process_change(self, change):
             device_app_meta = None
             if device_id and app_id:
                 device_app_meta = DeviceAppMeta(app_id=app_id, build_id=build_id, last_sync=sync_date)
-            mark_last_synclog(domain, user, app_id, build_id, sync_date, device_id, device_app_meta)
+            mark_last_synclog(domain, user, app_id, build_id, sync_date, sync_date, device_id, device_app_meta)
 
 
-def mark_last_synclog(domain, user, app_id, build_id, sync_date, device_id,
-        device_app_meta, commcare_version=None, build_profile_id=None, save_user=True):
+def mark_last_synclog(domain, user, app_id, build_id, sync_date, latest_build_date, device_id,
+        device_app_meta, commcare_version=None, build_profile_id=None, save_user=True,
+        build_update_date=None):
     version = None
     if build_id:
         version = get_version_from_build_id(domain, build_id)
 
-    local_save = update_last_sync(user, app_id, sync_date, version)
+    local_save = False
+    if sync_date:
+        # sync_date could be null if this is called from a heartbeat request
+        update_last_sync(user, app_id, sync_date, version)
```

Comment:
This used to get called only if a sync_date was available. See https://github.com/dimagi/commcare-hq/pull/26305/files#diff-291dfd3802bf0daa51e598788d5ca13bL352